src/signalrclient/default_websocket_client.h

```diff
@@ -20,12 +20,13 @@ namespace signalr
     public:
         explicit default_websocket_client(const signalr_client_config& signalr_client_config = {}) noexcept;
 
-        void start(const std::string& url, transfer_format format, std::function<void(std::exception_ptr)> callback);
+        void start(const std::string& url, signalr::transfer_format format, std::function<void(std::exception_ptr)> callback);
```

Comment:
A WebSocket can switch between text an binary in the same connection, right? Could we make the `transfer_format` an argument to `send` and `receive` rather than `start`? I know SignalR doesn't need to ever switch mid-connection, but it seems like a cleaner to match what the protocol allows.

Reply:
A WebSocket can, but SignalR can't/won't.

In the other clients we pass the format into start as well
https://github.com/dotnet/aspnetcore/blob/c0a2024a54c848c85ddff460dbf3fb12b7898f70/src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs#L112

https://github.com/dotnet/aspnetcore/blob/c0a2024a54c848c85ddff460dbf3fb12b7898f70/src/SignalR/clients/ts/signalr/src/WebSocketTransport.ts#L37

Reply:
I know the other clients do it this way. I never liked that about the other clients either. Let's make this one a bit more flexible.

Reply:
Updated, with this change the transport has to figure out the format itself in the receive loop. At least for WebSockets this is fine